labs/src/main/scala/org/scalalabs/basic/lab03/OptionExercise.scala

```diff
@@ -37,6 +43,19 @@ object OptionExercise02 {
    * to convert a possible numeric String (e.g. Some("12")) to an integer
    */
   def totalPeopleInRooms(rooms: Map[Int, Option[String]]): Int = {
-    error("Fix me")
+    
+    //calculated number of people given a room 
+    def numPeopleInRoom(room:Option[String]): Int = {
+      room match {
+        case None => 0
+        case s:Option[String] => if (s.get == "locked") 0 else (augmentString)(s.get).toInt
+      }
+    }
+    
+    // sum up all the people per room, return sum
+    var sum = 0
+    rooms.values.foreach(room => sum += numPeopleInRoom(room))
```

Comment:
This is very imperative code; a more Scala-y option might be to call `map` on `rooms.values`